API/fleece/Fleece.h

```diff
@@ -1195,7 +1232,145 @@ while (NULL != (value = FLDictIterator_GetValue(&iter))) {
+         its only purpose is to let you store a value into it, using the `FLSlot_...` functions.
+
+         Since there are three ways to store a value into a collection (array set, array append,
+         dict set) and nine types of values that can be stored, that makes 27 setter functions.
+         For efficiency, these are declared as inlines that call one of three functions to acquire
+         a slot, and one of nine functions to store a value into it.
+
+         It's usually more convenient to use the typed functions like \ref FLMutableArray_SetInt,
+         but you might drop down to the lower level ones if you're creating an adapter between
+         Fleece and a different data model, such as Apple's Foundation classes. */
+
+    /** Returns an \ref FLSlot that refers to the given index of the given array.
+        You store a value to it by calling one of the nine `FLSlot_Set...` functions.
+        \warning You should immediately store a value into the `FLSlot`. Do not keep it around;
+                 any changes to the array invalidate it.*/
+    MUST_USE_RESULT
+    FLSlot FLMutableArray_Set(FLMutableArray NONNULL, uint32_t index) FLAPI;
+
+    /** Appends a null value to the array and returns an \ref FLSLot that refers to that position.
+        You store a value to it by calling one of the nine `FLSlot_Set...` functions.
+        \warning You should immediately store a value into the `FLSlot`. Do not keep it around;
+                 any changes to the array invalidate it.*/
+    MUST_USE_RESULT
+    FLSlot FLMutableArray_Append(FLMutableArray NONNULL) FLAPI;
+
+    /** Returns an \ref FLSlot that refers to the given key/value pair of the given dictionary.
+        You store a value to it by calling one of the nine `FLSlot_Set...` functions.
+        \warning You should immediately store a value into the `FLSlot`. Do not keep it around;
+                 any changes to the dictionary invalidate it.*/
+    MUST_USE_RESULT
+    FLSlot FLMutableDict_Set(FLMutableDict FL_NONNULL, FLString key) FLAPI;
+
+
+    void FLSlot_SetNull(FLSlot NONNULL) FLAPI;             ///< Stores a JSON null into a slot.
+    void FLSlot_SetBool(FLSlot NONNULL, bool) FLAPI;       ///< Stores a boolean into a slot.
+    void FLSlot_SetInt(FLSlot NONNULL, int64_t) FLAPI;     ///< Stores an integer into a slot.
+    void FLSlot_SetUInt(FLSlot NONNULL, uint64_t) FLAPI;   ///< Stores an unsigned int into a slot.
+    void FLSlot_SetFloat(FLSlot NONNULL, float) FLAPI;     ///< Stores a `float` into a slot.
+    void FLSlot_SetDouble(FLSlot NONNULL, double) FLAPI;   ///< Stores a `double` into a slot.
+    void FLSlot_SetString(FLSlot NONNULL, FLString) FLAPI; ///< Stores a UTF-8 string into a slot.
+    void FLSlot_SetData(FLSlot NONNULL, FLSlice) FLAPI;    ///< Stores a data blob into a slot.
+    void FLSlot_SetValue(FLSlot NONNULL, FLValue) FLAPI;   ///< Stores an FLValue into a slot.
+
+
+    // implementations of the inline methods declared earlier:
```

Comment:
Since they are inline it is odd to me to see them forward declared.  Is there a technical reason for it, or is it just organizational?  I'm wondering if the forward declarations need to be there.

Reply:
Organizational, to keep the header cleaner for people reading it.